### DISCUSSION TITLE
How to create a Signal TLS proxy and integrate it into a running NGINX Proxy Manager?

### DISCUSSION THREAD

**Discussion by ErikUden** [NginxProxyManager/nginx-proxy-manager#1738] [Q&A]
Hello there!
I've recently stumbled upon this great article: https://community.signalusers.org/t/howto-integrate-signal-tls-proxy-into-existing-nginx-lets-encrypt-setup/27443

It shows how to integrate a Signal TLS proxy into a _running_ NGINX environment. 
This is important, because the [official documentation](https://github.com/signalapp/Signal-TLS-Proxy) is literally just one page and only explains how to create a Signal proxy on an entirely new NGINX reverse proxy environment on a VPS. Now, I'd wish to help my friends stay connected during conversations with a Signal proxy, but I really don't feel like spending the money for a VPS when I have a running, fast, high-bandwidth network with a perfectly working version of the NGINX Proxy Manager by jc21 running. 

However, I am very much a novice and cannot make much of the article mentioned above. As I understand it, the NGINX Proxy Manager by jc21 does not have a simple /etc/nginx/nginx.conf file to edit, right? That is why I was very confused as to what I would need to do to get this working! I know when I SSH into the docker container such a file exists, but I do not think I should just easily edit it like on any NGINX environment, so I wanted to ask here first!

Additionally, the [article](https://community.signalusers.org/t/howto-integrate-signal-tls-proxy-into-existing-nginx-lets-encrypt-setup/27443) mentions that the --with-stream argument would need to be set to "dynamic" or built this way, however, when I ran `nginx -V` whilst being SSH'ed into my NGINX Proxy Manager container I yielded the following result:

> configure arguments: --prefix=/etc/nginx/nginx --with-cc-opt=-O2 --add-module=../ngx_devel_kit-0.3.1 --add-module=../echo-nginx-module-0.62 --add-module=../xss-nginx-module-0.06 --add-module=../ngx_coolkit-0.2 --add-module=../set-misc-nginx-module-0.32 --add-module=../form-input-nginx-module-0.12 --add-module=../encrypted-session-nginx-module-0.08 --add-module=../srcache-nginx-module-0.32 --add-module=../ngx_lua-0.10.19 --add-module=../ngx_lua_upstream-0.07 --add-module=../headers-more-nginx-module-0.33 --add-module=../array-var-nginx-module-0.05 --add-module=../memc-nginx-module-0.19 --add-module=../redis2-nginx-module-0.15 --add-module=../redis-nginx-module-0.3.7 --add-module=../rds-json-nginx-module-0.15 --add-module=../rds-csv-nginx-module-0.09 --add-module=../ngx_stream_lua-0.0.9 --with-ld-opt=-Wl,-rpath,/etc/nginx/luajit/lib --sbin-path=/usr/sbin/nginx --modules-path=/usr/lib/nginx/modules --conf-path=/etc/nginx/nginx.conf --error-log-path=/var/log/nginx/error.log --http-log-path=/var/log/nginx/access.log --pid-path=/var/run/nginx.pid --lock-path=/var/run/nginx.lock --http-client-body-temp-path=/var/cache/nginx/client_temp --http-proxy-temp-path=/var/cache/nginx/proxy_temp --http-fastcgi-temp-path=/var/cache/nginx/fastcgi_temp --http-uwsgi-temp-path=/var/cache/nginx/uwsgi_temp --http-scgi-temp-path=/var/cache/nginx/scgi_temp --user=nginx --group=nginx --with-compat --with-threads --with-http_addition_module --with-http_auth_request_module --with-http_dav_module --with-http_flv_module --with-http_gunzip_module --with-http_gzip_static_module --with-http_mp4_module --with-http_random_index_module --with-http_realip_module --with-http_secure_link_module --with-http_slice_module --with-http_ssl_module --with-http_stub_status_module --with-http_sub_module --with-http_v2_module --with-mail --with-mail_ssl_module --with-stream --with-stream_realip_module --with-stream_ssl_module --with-stream_ssl_preread_module --with-stream

So, I am not certain what the attribute "--with-stream" is set to, but it seems to be set to nothing. 

So, I just wanted to ask the community for help. Any chance to get this running in my docker container without eliminating all my other sites running on here, or not?

Any answer is highly appreciated!

**Comment by Ririshi** [comment]
Did you ever get this to work? I'm struggling with the same issue right now.
